admin-dev/themes/new-theme/js/pages/product/components/generator/CombinationGenerator.vue

```diff
@@ -82,9 +85,10 @@
   import AttributesSelector from '@pages/product/components/generator/AttributesSelector.vue';
   import isSelected from '@pages/product/mixins/is-attribute-selected';
   import {getAllAttributeGroups} from '@pages/product/services/attribute-groups';
-  import Modal from '@vue/components/Modal.vue';
+  import Modal from '@PSVue/components/Modal.vue';
+  import {defineComponent} from 'vue';
   import ProductEventMap from '@pages/product/product-event-map';
-  import {Attribute, AttributeGroup} from '@pages/product/types';
+  import {Attribute, AttributeGroup} from '.';
```

Comment:
Why not keep:
```suggestion
  import {Attribute, AttributeGroup} from '@pages/product/types';
```

Or if it bothers you to have one file in common between all the product components (and the types are not shared actually) why not creating a type file specific to this generator component But in any case, keeping the types separated seems cleaner, it actually seems to be what is done in the TypeScript libraries where you find a dedicated `type.d.ts` file (or something similar)